tests/unit/algorithms/test_corrector.py

```diff
@@ -0,0 +1,1186 @@
+                subset="train",
+                annotations=[
+                    Bbox(0, 0, 100, 100, id=1, label=0),
+                    Bbox(0, 0, np.inf, 10, id=2, label=1),
+                ],
+            ),
+            DatasetItem(
+                id="missing_label_0",
+                subset="train",
+            ),
+            DatasetItem(
+                id="normal_test_0",
+                subset="test",
+                annotations=[
+                    Bbox(0, 0, 100, 100, id=1, label=1),
+                    Bbox(0, 0, 100, 100, id=2, label=2),
+                ],
+            ),
+        ],
+        categories={
+            AnnotationType.label: LabelCategories.from_iterable(
+                "label_" + str(label) for label in range(4)
+            ),
+        },
+    )
+
+
+@pytest.fixture
+def fxt_refined_det_dataset():
+    return Dataset.from_iterable(
+        [
+            DatasetItem(
+                id="normal_train_0",
+                subset="train",
+                annotations=[
+                    Bbox(0, 0, 100, 100, id=1, label=1),
+                ],
+            ),
+            DatasetItem(
+                id="normal_train_1",
+                subset="train",
+                annotations=[
+                    Bbox(0, 0, 100, 100, id=1, label=1),
+                    Bbox(50, 50, 100, 100, id=2, label=2),
+                ],
+            ),
+            DatasetItem(
+                id="undefined_label_0",
+                subset="train",
+                annotations=[
+                    Bbox(0, 0, 100, 100, id=1, label=0),
+                    Bbox(0, 0, 100, 100, id=2, label=100),
+                ],
+            ),
+            DatasetItem(
+                id="negative_len_0",
+                subset="train",
+                annotations=[
+                    Bbox(0, 0, 100, 100, id=2, label=1),
+                ],
+            ),
+            DatasetItem(
+                id="negative_len_1",
+                subset="train",
+                annotations=[
+                    Bbox(0, 0, 100, 100, id=1, label=0),
+                ],
+            ),
+            DatasetItem(
+                id="invalid_0",
+                subset="train",
+                annotations=[
+                    Bbox(0, 0, 100, 100, id=1, label=0),
+                ],
+            ),
+            DatasetItem(
+                id="normal_test_0",
+                subset="test",
+                annotations=[
+                    Bbox(0, 0, 100, 100, id=1, label=1),
+                    Bbox(0, 0, 100, 100, id=2, label=2),
+                ],
+            ),
+        ],
+        categories={
+            AnnotationType.label: LabelCategories.from_iterable(
+                ["label_" + str(label) for label in range(4)] + ["100"]
+            ),
+        },
+    )
+
+
+class ValidationCorrectionTest:
+    @pytest.mark.parametrize(
+        "fxt_original_dataset, fxt_validator, fxt_val_args, fxt_refined_dataset",
```

Comment:
`fxt_val_args` seems empty for all test cases. Is it necessary?

Reply:
I will remove. Thank you.